test/03_device-state.ts

```diff
@@ -38,296 +38,315 @@ configMock['API_HEARTBEAT_STATE_TIMEOUT_SECONDS'] = TIMEOUT_SEC;
-
-				const pollInterval = await stateMock.getPollInterval(device.uuid);
-				expect(pollInterval).to.equal(POLL_MSEC * stateMock.POLL_JITTER_FACTOR);
-
-				await supertest(admin)
-					.delete(
-						`/${version}/device_config_variable?$filter=name eq 'RESIN_SUPERVISOR_POLL_INTERVAL' and device eq ${device.id}`,
-					)
-					.expect(200);
-			});
+				let deviceUserRequestedState: fakeDevice.Device;
 
-			it('Should see the default value if the application-specific value is less than it', async () => {
-				await supertest(admin)
-					.patch(
-						`/${version}/application_config_variable?$filter=name eq 'RESIN_SUPERVISOR_POLL_INTERVAL' and application eq ${applicationId}`,
-					)
-					.send({
-						value: `${POLL_MSEC - 200}`,
-					})
-					.expect(200);
-
-				const pollInterval = await stateMock.getPollInterval(device.uuid);
-				expect(pollInterval).to.equal(POLL_MSEC * stateMock.POLL_JITTER_FACTOR);
-
-				await supertest(admin)
-					.delete(
-						`/${version}/application_config_variable?$filter=name eq 'RESIN_SUPERVISOR_POLL_INTERVAL' and application eq ${applicationId}`,
-					)
-					.expect(200);
-			});
-		});
+				const stateChangeEventSpy = sinon.spy();
+				before(async () => {
+					deviceUserRequestedState = await fakeDevice.provisionDevice(
+						admin,
+						applicationId,
+					);
 
-		describe('Event Tracking', () => {
-			const devicePollInterval =
-				Math.ceil((POLL_MSEC * stateMock.POLL_JITTER_FACTOR) / 1000) * 1000;
+					stateMock.getInstance().on('change', (args) => {
+						if (
+							![device.uuid, deviceUserRequestedState.uuid].includes(args.uuid)
+						) {
+							return;
+						}
 
-			let deviceUserRequestedState: fakeDevice.Device;
+						stateChangeEventSpy(args);
+					});
+				});
 
-			const stateChangeEventSpy = sinon.spy();
-			before(async () => {
-				deviceUserRequestedState = await fakeDevice.provisionDevice(
-					admin,
-					applicationId,
-				);
+				it('Should see the stats event emitted more than three times', async () => {
+					const statsEventSpy = sinon.spy();
+					stateMock.getInstance().on('stats', statsEventSpy);
 
-				stateMock.getInstance().on('change', (args) => {
-					if (
-						![device.uuid, deviceUserRequestedState.uuid].includes(args.uuid)
-					) {
-						return;
-					}
+					await waitFor({ checkFn: () => statsEventSpy.callCount >= 3 });
 
-					stateChangeEventSpy(args);
+					stateMock.getInstance().off('stats', statsEventSpy);
 				});
-			});
-
-			it('Should see the stats event emitted more than three times', async () => {
-				const statsEventSpy = sinon.spy();
-				stateMock.getInstance().on('stats', statsEventSpy);
 
-				await waitFor({ checkFn: () => statsEventSpy.callCount >= 3 });
+				[
+					{
+						tokenType: 'device API Key',
+						getActor: () => device,
+						heartbeatAfterGet: DeviceOnlineStates.Online,
+						getDevice: () => device,
+						getStateV2: () =>
+							fakeDevice.getState(device, device.uuid, stateVersion as any),
```

Comment:
I could not figure out how to avoid the `stateVersion as any`, without it everything I tried got `"v2" | "v3" is not compatible with "v2"` 😒